tests/models/test_taskinstance.py

```diff
@@ -992,3 +994,116 @@ def success_handler(self, context):
         self.assertEqual(cw.task_state_in_callback, State.RUNNING)
         ti.refresh_from_db()
         self.assertEqual(ti.state, State.SUCCESS)
+
+    @staticmethod
+    def _test_previous_dates_setup(schedule_interval: Union[str, datetime.timedelta, None],
+                                   catchup: bool) -> dict:
+        dag_id = 'test_previous_dates'
+        dag = models.DAG(dag_id=dag_id, schedule_interval=schedule_interval, catchup=catchup)
+        task = DummyOperator(task_id='task', dag=dag, start_date=DEFAULT_DATE)
+
+        def get_test_ti(session, execution_date: str, state: str) -> TI:
+            dag.create_dagrun(
+                run_id='scheduled__{}'.format(execution_date),
+                state=state,
+                execution_date=pendulum.parse(execution_date),
+                start_date=pendulum.utcnow(),
+                session=session
+            )
+            ti = TI(task=task, execution_date=pendulum.parse(execution_date))
```

Comment:
It obviously isn't a requirement as the tests are passing but
```suggestion
            ti = TI(task=task, dag_id=dag.id, execution_date=pendulum.parse(execution_date))
```

Reply:
TI does not take dag_id arg it gets it from task

Reply:
ashb: 0:
dstandish: 2

😁